node/hack/repair_terra/repair.go

```diff
@@ -0,0 +1,369 @@
+		if !ok {
+			log.Println("wormhole event does not have a sequence field", zap.String("tx_hash", txHash), zap.String("attributes", attributes.String()))
+			continue
+		}
+		blockTime, ok := mappedAttributes["message.block_time"]
+		if !ok {
+			log.Println("wormhole event does not have a block_time field", zap.String("tx_hash", txHash), zap.String("attributes", attributes.String()))
+			continue
+		}
+
+		senderAddress, err := StringToAddress(sender)
+		if err != nil {
+			log.Println("cannot decode emitter hex", zap.String("tx_hash", txHash), zap.String("value", sender))
+			continue
+		}
+		txHashValue, err := StringToHash(txHash)
+		if err != nil {
+			log.Println("cannot decode tx hash hex", zap.String("tx_hash", txHash), zap.String("value", txHash))
+			continue
+		}
+		payloadValue, err := hex.DecodeString(payload)
+		if err != nil {
+			log.Println("cannot decode payload", zap.String("tx_hash", txHash), zap.String("value", payload))
+			continue
+		}
+
+		blockTimeInt, err := strconv.ParseInt(blockTime, 10, 64)
+		if err != nil {
+			log.Println("blocktime cannot be parsed as int", zap.String("tx_hash", txHash), zap.String("value", blockTime))
+			continue
+		}
+		nonceInt, err := strconv.ParseUint(nonce, 10, 32)
+		if err != nil {
+			log.Println("nonce cannot be parsed as int", zap.String("tx_hash", txHash), zap.String("value", blockTime))
+			continue
+		}
+		sequenceInt, err := strconv.ParseUint(sequence, 10, 64)
+		if err != nil {
+			log.Println("sequence cannot be parsed as int", zap.String("tx_hash", txHash), zap.String("value", blockTime))
+			continue
+		}
+		messagePublication := &common.MessagePublication{
+			TxHash:           txHashValue,
+			Timestamp:        time.Unix(blockTimeInt, 0),
+			Nonce:            uint32(nonceInt),
+			Sequence:         sequenceInt,
+			EmitterChain:     vaa.ChainIDTerra,
+			EmitterAddress:   senderAddress,
+			Payload:          payloadValue,
+			ConsistencyLevel: 0, // Instant finality
+		}
+		msgs = append(msgs, messagePublication)
+	}
+	return msgs
+}
+
+// StringToAddress convert string into address
+func StringToAddress(value string) (vaa.Address, error) {
+	var address vaa.Address
+	res, err := hex.DecodeString(value)
+	if err != nil {
+		return address, err
+	}
+	copy(address[:], res)
+	return address, nil
+}
+
+// StringToHash convert string into transaction hash
+func StringToHash(value string) (eth_common.Hash, error) {
+	var hash eth_common.Hash
+	res, err := hex.DecodeString(value)
+	if err != nil {
+		return hash, err
+	}
+	copy(hash[:], res)
+	return hash, nil
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+
+	missingMessages := make(map[uint64]bool)
+
+	conn, err, admin := getAdminClient(ctx, *adminRPC)
+	defer conn.Close()
+	if err != nil {
+		log.Fatalf("failed to get admin client: %v", err)
+	}
+
+	var TerraEmitter = struct {
+		ChainID vaa.ChainID
+		Emitter string
+	}{vaa.ChainIDTerra, "0000000000000000000000007cf7b764e38a0a5e967972c1df77d432510564e2"}
```

Comment:
Nit: generally best to define things like that as a package level const

Reply:
Moved.